test/unit/CollateralPool.js

```diff
@@ -252,6 +300,42 @@ describe('Collateral Pool', async function () {
       expect(await token.balanceOf(alice.address)).to.be.equal(10)
     })
 
+    it('Should work to withdraw to other address and from owner != sender', async function () {
+      const { alice, bob, cToken, pool, token } = await loadFixture(deploy)
+
+      await token.mint(bob.address, 1000)
+      await token.connect(bob).approve(pool.address, 1000)
+
+      // Overloading Ethers-v6
+      expect(await pool.connect(bob)['deposit(uint256)'](1000)).to.emit(pool, 'Deposit')
+      expect(await cToken.balanceOf(bob.address)).to.be.equal(1000)
+      expect(await token.balanceOf(bob.address)).to.be.equal(0)
+      expect(await token.balanceOf(alice.address)).to.be.equal(0)
+
+      await cToken.connect(bob).approve(alice.address, 10)
+
+      // Overloading Ethers-v6
+      expect(await pool.connect(alice)['withdraw(uint256,address,address)'](10, alice.address, bob.address)).to.emit(pool, 'Withdraw')
+      expect(await cToken.balanceOf(bob.address)).to.be.equal(990)
+      expect(await token.balanceOf(bob.address)).to.be.equal(0)
+      expect(await token.balanceOf(alice.address)).to.be.equal(10)
+      expect(await cToken.allowance(bob.address, alice.address)).to.be.equal(0)
```

Comment:
let's test the "full" example and  change this for "bob allow Alice to spend and **treasury** receive the tokens" =D (treasury is global account)

```suggestion
      expect(await pool.connect(alice)['withdraw(uint256,address,address)'](10, treasury.address, bob.address)).to.emit(pool, 'Withdraw')
      expect(await cToken.balanceOf(bob.address)).to.be.equal(990)
      expect(await token.balanceOf(bob.address)).to.be.equal(0)
      expect(await token.balanceOf(alice.address)).to.be.equal(0)
      expect(await token.balanceOf(treasury.address)).to.be.equal(10)
      expect(await cToken.allowance(bob.address, alice.address)).to.be.equal(0)
```